test/new-e2e/tests/agent-runtimes/forwarder/nss_failover_test.go

```diff
@@ -199,6 +225,12 @@ func (v *multiFakeIntakeSuite) requireIntakeIsUsed(intake *fi.Client, intakeMaxW
 		require.NoError(t, err)
 		assert.NotEmpty(t, logs)
 
+		// check traces
+		runUDSTraceGenerator(v.Env().Host, "test", "extratags")
```

Comment:
You don't use the return value to cleanup, is that on purpose ?

Reply:
Hm good catch. It doesn't really matter but I'll change the `runUDSTraceGenerator` to do the cleanup directly instead.

Reply:
Actually, after further test, I can't immediately tear it down or no traces are emitted, I've thus re-introduced the cleanup return value and I'm using it now.